docs/api-reference/service-status-api.md

```diff
@@ -23,154 +23,1002 @@ sidebar_label: Service status
-client on the node has started to receive data from the ZooKeeper cluster and it's expected that all segments and other
-nodes will be discovered by this node timely from this point.
 
-`GET /status/selfDiscovered`
+### Get node self-discovery status
 
-Similar to `/status/selfDiscovered/status`, but returns 200 OK response with empty body if the node has discovered itself
-and 503 SERVICE UNAVAILABLE if the node hasn't discovered itself yet. This endpoint might be useful because some
-monitoring checks such as AWS load balancer health checks are not able to look at the response body.
+#### URL
+<code class="getAPI">GET</code> `/status/selfDiscovered`
 
-## Master server
+Retrieves a status code to indicate if a node discovered itself within the Druid cluster. This is similar to the `status/selfDiscovered/status` endpoint but relies on HTTP status codes alone. This is useful for certain monitoring checks such as AWS load balancer health checks that are unable to examine the response body.
 
-### Coordinator
+#### Responses
 
-#### Leadership
+<!--DOCUSAURUS_CODE_TABS-->
 
-`GET /druid/coordinator/v1/leader`
+<!--200 SUCCESS-->
+<br/>
+*Successfully retrieved node status*  
 
-Returns the current leader Coordinator of the cluster.
+<!--503 SERVICE UNAVAILABLE-->
+<br/>
+*Unsuccessful node self-discovery*  
+<!--END_DOCUSAURUS_CODE_TABS-->
 
-`GET /druid/coordinator/v1/isLeader`
+#### Sample request
+
+<!--DOCUSAURUS_CODE_TABS-->
+
+<!--cURL-->
+```shell
+curl "{domain}/status/selfDiscovered"
+```
+<!--HTTP-->
+```http
+GET /status/selfDiscovered HTTP/1.1
+Host: {domain}
+```
+
+<!--END_DOCUSAURUS_CODE_TABS-->
+
+#### Sample response
+A successful response to this endpoint results in an empty response body.
+
+## Coordinator
+
+### Get leader address
+
+#### URL
+
+<code class="getAPI">GET</code> `/druid/coordinator/v1/leader`
+
+Retrieves the address of the current leader Coordinator of the cluster.
+
+#### Responses
+
+<!--DOCUSAURUS_CODE_TABS-->
+
+<!--200 SUCCESS-->
+<br/>
+*Successfully retrieved leader Coordinator address*  
+<!--END_DOCUSAURUS_CODE_TABS-->
+
+---
+
+#### Sample request
+
+<!--DOCUSAURUS_CODE_TABS-->
+<!--cURL-->
+```shell
+curl "{domain}/druid/coordinator/v1/leader"
+```
+<!--HTTP-->
+```http
+GET /druid/coordinator/v1/leader HTTP/1.1
+Host: {domain}
+```
+<!--END_DOCUSAURUS_CODE_TABS-->
+
+#### Sample response
+<details>
+  <summary>Click to show sample response</summary>
+  ```json
+  http://localhost:8081
+  ```
+</details>
+
+### Get leader status
+
+#### URL
+<code class="getAPI">GET</code> `/druid/coordinator/v1/isLeader`
+
+Retrieves a JSON object with a `leader` key. The value can be `true` or `false`, indicating if this server is the current leader Coordinator of the cluster. This is suitable for use as a load balancer status check if you only want the active leader to be considered in-service at the load balancer.
```

Comment:
Would you need to run this against all the individual Coordinator endpoints?

Reply:
I think with the /leader/ endpoint, there isn't a need to run it against all coordinators since /leader/ will check all of them and /isLeader/ is for a particular one.

Reply:
No, this is a per process/node query.